### PR TITLE
Add -copy_extensions option to x509 utility

### LINKED ISSUE #3638: Support "copy_extensions" also with x509 CSR signing
While already supported with "openssl ca", basic signing does not support the "copy_extension" mode. Basic signing might be neccessary when the "openssl ca" magic is too much and cannot be turned off in certain usecases. Thus when using "openssl x509" instead, from each CSR, a openssl.config has to be created manually by duplicating the CSR fields before signing, which makes it even more risky and error prone than using the "copy_extensions".

It would be nice to support the existing "copy_extensions = copy" feature also in for "openssl x509". (It would be even more nice, if it would allow "... = copy:subjectAltName", but that is another story ...)

### PR DESCRIPTION
**Pull request by ca6aqa**
-copy_extensions option specifies how extensions from SOURCE --
certificate request (if -req option present) or certificate
(if -x509toreq option present) -- should be handled when transferred to
TARGET -- certificate or certificate request, accordingly.
If set to none or if the option is not present then extensions
are ignored and not copied from SOURCE to TARGET.
If set to copy then any extension present in the SOURCE but not yet
present in TARGET is copied from SOURCE to TARGET.
If set to copyall then all extensions in the SOURCE are copied to
TARGET; if the extension is already present in TARGET (e.g. as a result
of -extfile option being specified), it is overwritten.

A new public function X509_to_X509_REQ_ex is also added. It differs
from the X509_to_X509_REQ function in that it takes flag as an
additional parameter, which determines whether it is necessary
to copy extensions from the certificate to the request.

Fixes #3638 and #6481.

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: crypto/x509/x509_req.c [hunk 1]
```diff
@@ -50,6 +50,12 @@ X509_REQ *X509_to_X509_REQ(X509 *x, EVP_PKEY *pkey, const EVP_MD *md)
     if (!i)
         goto err;
 
+    if (flags == XTXR_FLAG_COPY_EXTENSIONS) {
+        const STACK_OF(X509_EXTENSION) *extlist = X509_get0_extensions(x);
```
Comment (FdaSilvaYY) [inline crypto/x509/x509_req.c]: style : blank line after declaration.

File: CHANGES [hunk 2]
```diff
@@ -9,6 +9,10 @@
 
  Changes between 1.1.1 and 3.0.0 [xx XXX xxxx]
 
+  *) Added X509_to_X509_REQ_ex function. It is allows to copy extensions
```
Comment (FdaSilvaYY) [inline CHANGES]: ` It allows copy of extensions ..` will be better IMHO.

File: crypto/x509/x509_req.c (hunk 1 above)
Comment (ca6aqa) [reply crypto/x509/x509_req.c]: fixed

File: CHANGES (hunk 2 above)
Comment (ca6aqa) [reply CHANGES]: fixed

**Comment by ionphractal** [comment]
Why was this closed? @ca6aqa Is there an alternative?

**Comment by nyetwurk** [comment]
bump

**Comment by Mironenko** [comment]
> 
> 
> Why was this closed? @ca6aqa Is there an alternative?

I think the main problem is that it's been a quarter of a year since PR had been proposed and fixed after the review, but there was no further attention from the maintainers.

**Pull request closed without merging**